notebook/notebookapp.py

```diff
@@ -666,6 +667,14 @@ def _valdate_ip(self, proposal):
             value = u''
         return value
 
+    connect_host = Unicode(u'', config=True,
+        help=_("""Custom host to show in displayed URL.
+
+        Replace actual host, including protocol, address, port and base URL,
+        with the given value when displaying URL to the users. Do not change
```

Comment:
We probably should clarify this docstring a bit. As it reads now, my understanding is that the connect_host would be displayed to the user, but the notebook would use the actual underlying connection URL.

Reply:
Funnily I had a *very* similar docstring amending in store :) I merged the two. Tell me if it seems clearer.

Reply:
Updated. tell me if it's still not clear enough (your understanding was correc btw :wink:)